exp/api/v1beta1/azuremachinepool_types.go

```diff
@@ -157,12 +157,6 @@ type (
 		// +kubebuilder:default={type: "RollingUpdate", rollingUpdate: {maxSurge: 1, maxUnavailable: 0, deletePolicy: Oldest}}
 		Strategy AzureMachinePoolDeploymentStrategy `json:"strategy,omitempty"`
 
-		// NodeDrainTimeout is the total amount of time that the controller will spend on draining a node.
-		// The default value is 0, meaning that the node can be drained without any time limitations.
-		// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
-		// +optional
-		NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
```

Comment:
note: this is a breaking change

it's okay since AzureMachinePool is still an experimental type but it needs to be clearly called out in the release notes and we might want to just mark it as deprecate for now instead of deleting it to be safe

Reply:
If we leave it in, it will effectively do nothing. Does deprecating a field mean removing functionality in this way? I was under the impression that it meant it would no longer receive updates in favor of a new solution.

Reply:
Ideally a non-breaking deprecation would first warn about the field getting removed, and then remove it in a future release. In this case since it will actually no longer work and the API is still experimental, removing it makes most sense but it should be called out as a breaking change in the PR release notes.

https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md